src/Lecture2.hs

```diff
@@ -81,7 +93,9 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+evenLists :: [[a]] -> [[a]]
+evenLists = filter isEven where
+  isEven l = even . length $ l
```

Comment:
You can eta-reduce the function inside `where`

```suggestion
evenLists = filter isEven where
  isEven = even . length
```

And in this particular case, it's rather small so you can inline its definition as a next step.

```suggestion
evenLists = filter (even . length)
```